modules/heat_conduction/src/fvbcs/FVFunctorConvectiveHeatFluxBC.C

```diff
@@ -0,0 +1,49 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FVFunctorConvectiveHeatFluxBC.h"
+#include "Function.h"
+
+registerMooseObject("MooseApp", FVFunctorConvectiveHeatFluxBC);
+
+InputParameters
+FVFunctorConvectiveHeatFluxBC::validParams()
+{
+  InputParameters params = FVFluxBC::validParams();
+  params.addClassDescription(
+      "Convective heat transfer boundary condition with temperature and heat "
+      "transfer coefficent given by functors.");
+  params.addRequiredParam<MooseFunctorName>("T_wall", "Functor for wall temperature");
+  params.addRequiredParam<MooseFunctorName>("T_infinity", "Functor for far-field temperature");
+  params.addRequiredParam<MooseFunctorName>("heat_transfer_coefficient",
+                                            "Functor for heat transfer coefficient");
+  params.addRequiredParam<bool>("is_solid", "Whether this kernel acts on the solid temperature");
+
+  return params;
+}
+
+FVFunctorConvectiveHeatFluxBC::FVFunctorConvectiveHeatFluxBC(const InputParameters & parameters)
+  : FVFluxBC(parameters),
+    _T_wall(getFunctor<ADReal>("T_wall")),
+    _T_infinity(getFunctor<ADReal>("T_infinity")),
+    _htc(getFunctor<ADReal>("heat_transfer_coefficient")),
+    _is_solid(getParam<bool>("is_solid"))
+{
+}
+
+ADReal
+FVFunctorConvectiveHeatFluxBC::computeQpResidual()
+{
+  if (_is_solid)
```

Comment:
Some chime-in from computer-science educated people. 
Does it make a difference in runtime to either predefine `Real mult = is_solid ? -1 : 1` versus doing the `if (_is_solid)` every time this is called?

Reply:
The cost of the if condition check will be negligible compared to the functor evaluations that go on below this. But it would be nice to predefine the sign during construction and then this function could be a one-liner